src/pages/patientView/timeline2/VAFChart.tsx

```diff
@@ -205,14 +205,20 @@ export default class VAFChart extends React.Component<IVAFChartProps, {}> {
     }
 
     @computed get mutationToDataPoints() {
-        const map = new ComplexKeyMap<IColorPoint[]>();
+        const map = new ComplexKeyMap<IColorPoint[][]>();
```

Comment:
I personally think that the ComplexKeyMap is a bit overkill here. It does not help for readibility. What do you think of:

```
 @computed get mutationToLines() : {[mutation: string]: ILine[]}
```

used together with a function that creates a key bases on a mutation:

```
function getMutationKey(mutation: Mutation) {
   retrurn mutation.proteinChange + '_' + mutation.gene.hugoGeneSymbol;
}
```

The whole piece will look like:

```
     @computed get mutationToLines() : {[mutation: string]: ILine[]} {
        const mutationToLines: {[mutation: string]: ILine[]} = {};
        for (const lineData of this.props.lineData) {
            const key = getMutationKey(lineData.mutation);
            if (!mutationToLines[key])
                  mutationToLines[key] = [];
            mutationToLines[key].push(lineData);
        }
        return map;
    }
```

This can even be abbreviated to:

```
@computed get mutationToLines() : {[mutation: string]: ILine[]} {
      _.groupBy(this.props.lineData, (line: ILine) => getMutationKey(line.mutation));
}
```

So, a one-liner can replace 17 lines of code :)

Reply:
Fixed